pkg/reconciler/dependencybuild/controller.go

```diff
@@ -32,6 +36,25 @@ func SetupNewReconcilerWithManager(mgr ctrl.Manager, nonCachingClient client.Cli
 				return true
 			},
 		})).
-		Watches(&source.Kind{Type: &v1beta1.TaskRun{}}, &handler.EnqueueRequestForOwner{OwnerType: &v1alpha1.DependencyBuild{}, IsController: false}).
+		Watches(&source.Kind{Type: &v1beta1.TaskRun{}}, handler.EnqueueRequestsFromMapFunc(func(o client.Object) []reconcile.Request {
+			taskRun := o.(*v1beta1.TaskRun)
+
+			// check if the TaskRun is related to DependencyBuild
+			if taskRun.GetLabels() == nil {
+				return []reconcile.Request{}
+			}
+			_, ok := taskRun.GetLabels()[artifactbuild.TaskRunLabel]
```

Comment:
looking at the artifactbuild usage now @stuartwdouglas, should we filter out taskruns that have the taskrunlabel AND the artifactbuildidlabel (as those would be intended for the artifactbuild reconciler?

that said, I at least did not observe overt issues with the code like this